daemon/c_smartcard.c

```diff
@@ -1077,7 +1088,11 @@ c_smartcard_new(compartment_t *compartment)
 	smartcard->err_cb = NULL;
 	smartcard->err_cbdata = NULL;
 
+#ifdef SYSTEMD
+	smartcard->sock = sock_sd_listen_fd(SYSTEMD_SCD_CONTROL_SOCKET);
```

Comment:
this seems wrong we do not want the listening socket, we want to connect as client